### DISCUSSION TITLE
[V6] model count() method returning undefined

### DISCUSSION THREAD

**Discussion by holoyan** [adonisjs/core#4459] [Help]
`node ace make:model Product`

I follwed the  [docs](https://lucid.adonisjs.com/docs/select-query-builder#count) to get total amount of `products` but for some reason it's returnes undefined, not sure what I'm doing wrong

```typescript

  emitter.on('db:query', function (query) {
    console.log(query)
  })

  const products = await Product.query().count('*', 'total')
  console.log(products[0])
  console.log(products[0].total)


```

```
{
  duration: [ 0, 2218968 ],
  connection: 'postgres',
  inTransaction: false,
  model: 'Product',
  __knexUid: '__knexUid1',
  __knexTxId: undefined,
  method: 'select',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [],
  __knexQueryUid: '04VsT-YjS62iGqIHGx-Gd',
  sql: 'select count(*) as "total" from "products"',
  error: undefined
}
Product {
  modelOptions: { connection: 'postgres' },
  modelTrx: undefined,
  transactionListener: [Function: bound listener],
  fillInvoked: false,
  cachedGetters: {},
  forceUpdate: false,
  '$columns': {},
  '$attributes': {},
  '$original': {},
  '$preloaded': {},
  '$extras': { total: '2' },
  '$sideloaded': {},
  '$isPersisted': true,
  '$isDeleted': false,
  '$isLocal': false
}
undefined


```

**Comment by holoyan** [comment]
if I write `products[0].$extras.total` then it will work, but based on [docs](https://lucid.adonisjs.com/docs/select-query-builder#count) it should be 


> const users = await db
  .from('users')
  .count('* as total')
console.log(users[0].total)

**Reply by tomgobich** [reply]
That doc is making use of the Database Query Builder. The Model Query Builder [will always return instances of the model you're querying with](https://lucid.adonisjs.com/docs/model-query-builder). Anything not directly on the model, like aggregates, will be placed on the `$extras` object. So, [what you're describing is as expected](https://lucid.adonisjs.com/docs/serializing-models#serializing-extras-object)!

As Wswawan mentioned, you can, however, use `pojo()` to get through to the plain JavaScript objects, which typically match the Database Query Builder results.

```ts
const { total } = await Product.query().count('*', 'total').pojo().firstOrFail()
```

**Comment by wswawan** [comment]
Have you tried with pagination ?

```
const products = await Product.query().paginate(1)
const total = products.total
return total
```

Or

```
const total = await Product.query().count('* as total').pojo()
return total
```

**Comment by holoyan** [comment]
Thank you all
